src/gui/guiChatConsole.cpp

```diff
@@ -778,3 +815,17 @@ void GUIChatConsole::updatePrimarySelection()
 	std::string selected = wide_to_utf8(wselected);
 	Environment->getOSOperator()->copyToPrimarySelection(selected.c_str());
 }
+
+void GUIChatConsole::updateScrollbar(bool update_size)
+{
+	m_scrollbar->setMin(m_chat_backend->getConsoleBuffer().getTopScrollPos());
+	m_scrollbar->setMax(m_chat_backend->getConsoleBuffer().getBottomScrollPos());
+	m_scrollbar->setPos(m_chat_backend->getConsoleBuffer().getScrollPosition());
+	m_scrollbar->setVisible(m_scrollbar->getMin() != m_scrollbar->getMax());
+	m_scrollbar->setPageSize(m_fontsize.Y * m_chat_backend->getConsoleBuffer().getLineCount());
+
+	if (update_size) {
+		const core::rect<s32> rect (m_screensize.X - getScrollbarSize(Environment), 0, m_screensize.X, m_height);
```

Comment:
Duplicated from `GUIChatConsole::drawText()`. You could move this into a member function instead of a static one to calculate the entire rectangle there.